Add tutorials for working with the node terminus

michaeltchapman opened the pull request:
This patch adds some tutorials, presented as scripts, a
puppet file and a vagrant file, that demonstrate usage
of the scenario_node_terminus on a single node.

Commit:
Add tutorials for working with the node terminus

This patch adds some tutorials, presented as scripts, a
puppet file and a vagrant file, that demonstrate usage
of the scenario_node_terminus on a single node.

bodepd commented on tutorials/01install.sh:
```diff
@@ -0,0 +1,16 @@
+cd /etc/puppet/modules
+git clone https://github.com/puppetlabs/puppetlabs-stdlib stdlib
+git clone https://github.com/stephenrjohnson/puppetlabs-puppet puppet
+git clone -b havana https://github.com/CiscoSystems/puppet-apt apt
```
does this need to be this branch? Also, can you submit the extra cisco code as a patch to upstream?

michaeltchapman replied:
It didn't work with master, I don't recall why. I'll check.

And yep I'll do that.

michaeltchapman replied:
There doesn't seem to be an equivalent for this line which Cisco added:

Exec["apt_update"] -> Package <| |>

I've just put it in setup.pp and amended this patch, and now it works with the puppetlabs repo.

Commit:
Add markdown generation

This adds a short python script and a blurb
so that the tutorials can be converted to a
markdown file which is a bit easier to read
on github.

Commit:
Move comments to single line

Let line wrapping do it's thing, since we're
generating markdown and not writing things
for the terminal.
